node-hub/dora-magma/pyproject.toml

```
@@ -0,0 +1,44 @@
+[build-system]
+requires = ["poetry-core>=1.8.0"]
+build-backend = "poetry.core.masonry.api"
+
+[project]
+name = "dora-magma"
+version = "0.1.0"
+description = "Dora node for Microsoft Magma model"
+requires-python = ">=3.10"
+license = {text = "MIT"}
+readme = "README.md"
+authors = [
+  {name = "Munish Mummadi", email = "moneymindedmunish1@gmail.com"}
+]
+dependencies = [
+    "torch>=2.4.0",
+    "torchvision>=0.19",
+    "transformers>=4.0.0",
+    "pillow>=9.0.0",
+    "numpy>=1.26.0,<2.0.0",
+    "dora-rs>=0.3.9",
+    "accelerate>=1.3.0",
+]
+
+[dependency-groups]
+dev = [
+    "pytest>=8.1.1",
+    "ruff>=0.9.1",
+    "pytest-cov>=4.0.0",
+]
+
+[project.scripts]
+dora-magma = "dora_magma.main:main"
+
+[tool.ruff]
+exclude = ["dora-magma/dora_magma/Magma"]
+
+[tool.black]
+extend.exclude = "dora_magma/Magma"
```

```suggestion
exclude = ["dora_magma/Magma"]

[tool.black]
extend.exclude = "dora_magma/Magma"
```

Have you tried this?

---

Simply put. No, I haven't. I thought path relevancy won't be an issue. pushed a commit. will check it 💯

---

Unfortunately, path is the problem. It costed a night lol. Thank you for the feedback. By any chance, do you want me to add my little odd experience to the `node-hub/README.me`

---

Sure thing! On how to add a git dependency without breaking on linting is definitely not easy!

I think adding a comment within this Magma line as a reference could be great as well!

---

Noted!! As soon as I am done with this issue, I will push a patch for that readme.